src/main/java/com/minecolonies/coremod/entity/ai/basic/AbstractEntityAIStructure.java

```diff
@@ -515,7 +573,20 @@ private void requestMaterials()
                 final AbstractBuilding building = getOwnBuilding();
                 if (building instanceof BuildingBuilder)
                 {
-                    ((BuildingBuilder) building).addNeededResource(block, 1);
+                    if(((JobBuilder) job).getStructure().getBlockInfo().tileentityData != null)
+                    {
+                        List<ItemStack> itemList = new ArrayList<>();
+                        itemList.addAll(getItemStacksOfTileEntity(((JobBuilder) job).getStructure().getBlockInfo().tileentityData));
+
+                        for(ItemStack stack: itemList)
+                        {
+                            if(!isBlockFree(stack) && checkOrRequestItems(stack))
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Refactor this code to not nest more than 5 if/for/while/switch/try statements. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://home.kk-sc.de:9000/coding_rules#rule_key=squid%3AS134)